napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+# Error handling
+
+N-API uses both return values and exceptions for error handling.  The sections
+which follow explain the approach for each case.
+
+## return values
```

Comment:
Maybe "return" should be capitalized, although if we want to stay true to the syntax in C++, it should not. Then again, in the latter case perhaps it should be in backquotes.